modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala

```diff
@@ -205,25 +204,40 @@ object KafkaProducer {
     * produces record in batches, limiting the number of records
     * in the same batch using [[ProducerSettings#parallelism]].
     */
-  def pipe[F[_], K, V, P](
+  def pipe[F[_], K, V](
     settings: ProducerSettings[F, K, V]
   )(
     implicit F: Async[F],
     mk: MkProducer[F]
-  ): Pipe[F, ProducerRecords[P, K, V], ProducerResult[P, K, V]] =
+  ): Pipe[F, ProducerRecords[K, V], ProducerResult[K, V]] =
     records => stream(settings).flatMap(pipe(settings, _).apply(records))
 
+  def pipeWithPassthrough[F[_], K, V, P](
```

Comment:
Actually I'm not sure that we need this.

Reply:
Oh, can user code be rewritten to do the same thing with `pipe`?

Reply:
I agree, we shouldn't keep adding pipe combinators. We can keep `pipe` but if you need something more involved, then you can use the non-pipe combinators.